dotCMS/src/main/java/com/dotcms/vanityurl/util/VanityUrlUtil.java

```diff
@@ -0,0 +1,41 @@
+package com.dotcms.vanityurl.util;
+
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+import com.dotmarketing.util.Logger;
+
+/**
+ * This class provide some utility methods to interact with
+ * Vanity URLs
+ *
+ * @author oswaldogallango
+ * @version 4.2.0
+ * @since June 12, 2017
+ */
+public class VanityUrlUtil {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A class which only has private constructors should be final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)